[docs] Document WebP support for emoji requests (#7568)
hammerandchisel/discord-api-docs: docs/resources/emoji.mdx

Emojis with subscription roles are considered premium emoji, and count toward a separate limit of 25.  
Emojis cannot be converted between normal and premium after creation.

###### Emoji Formats

> Emojis can be uploaded as JPEG, PNG, GIF, WebP, and AVIF formats. All emojis (regardless of original format) can be served as WebP. We highly recommend that developers request emojis as WebP for maximum performance and compatibility. The Discord client uses WebP for all emojis displayed in-app.

Reviewer: Rather than "request" emoji, should it be "submit emoji" (across the board)?

"request" feels like you are asking someone to submit something to me. Unless I'm missing a chunk of context, which is possible.

Reviewer 2: I think that is referring to fetching the images that already exist, not uploading new emoji

Reviewer: Maybe a silly question, since I don't know this area as well as others -- how are you requesting the emoji? Is there an API for requesting an emoji from somewhere? (that's what it reads like to me).

If not, I'd lean more towards something like:

```suggestion
Emojis can be uploaded as JPEG, PNG, GIF, WebP, and AVIF formats. All emojis (regardless of original format) can be served as WebP. We highly recommend that developers choose WebP emoji for maximum performance and compatibility. The Discord client uses WebP for all emojis displayed in-app.
```

Since that clears any ambiguity. WDYT?

Author: You're right that the wording is ambiguous. 'Request' here refers to when developers fetch emoji (`GET` requests), not when they upload them.  I'll update the text to: 'We highly recommend that developers use the `.webp` extension when fetching emojis so they're rendered as WebP for maximum  performance and compatibility.' This makes it clear we're talking about the file extension used when retrieving existing emojis.

Reviewer: 👍🏻 love it. LGTM!

Welp, looks like it fails on the `#emoji-object-emoji-formats` 🤔 I wonder why that is.

Reviewer: ...and you worked it out faster than I did. Full links required. `[Emoji Formats](/docs/resources/emoji#emoji-object-emoji-formats)`

Emojis and animated emojis have a maximum file size of 256 KiB. Attempting to upload an emoji larger than this limit will fail and return 400 Bad Request and an error message, but not a [JSON status code](/docs/topics/opcodes-and-status-codes#json).
:::

:::info

> We highly recommend requesting emojis as WebP for maximum performance and compatibility. See the Emoji Formats section above for more details.

Reviewer: Nit: "Emoji Formats" have it link to the header above, so it's easy to get to.

```suggestion
We highly recommend requesting emojis as WebP for maximum performance and compatibility. See the [Emoji Formats](#emoji-object-emoji-formats) section above for more details.
```

Emojis and animated emojis have a maximum file size of 256 KiB. Attempting to upload an emoji larger than this limit will fail and return 400 Bad Request and an error message, but not a [JSON status code](/docs/topics/opcodes-and-status-codes#json).
:::

:::info

> We highly recommend requesting emojis as WebP for maximum performance and compatibility. See the Emoji Formats section above for more details.

Reviewer: Nit: "Emoji Formats" have it link to the header above, so it's easy to get to.

```suggestion
We highly recommend requesting emojis as WebP for maximum performance and compatibility. See the [Emoji Formats](#emoji-object-emoji-formats) section above for more details.
```
